airflow/executors/kubernetes_executor.py

```diff
@@ -68,6 +73,18 @@
 ALL_NAMESPACES = "ALL_NAMESPACES"
 POD_EXECUTOR_DONE_KEY = "airflow_executor_done"
 
+if sys.version_info >= (3, 9):
+    from functools import cache
+else:
+    from functools import lru_cache
+
+    cache = lru_cache(maxsize=None)
```

Comment:
This is already available in `airflow.compar.functools`.

Reply:
Let me try to reuse that